components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3692,5 +3249,628 @@ subroutine get_time_space_phys_variables( &
+
+   real(rtype), intent(in), dimension(kts:kte) :: rho
+   real(rtype), intent(in), dimension(kts:kte) :: inv_rho
+   real(rtype), intent(in), dimension(kts:kte) :: rhofacr  
+   real(rtype), intent(in), dimension(kts:kte) :: rcldm 
+   real(rtype), intent(in), dimension(kts:kte) :: inv_dzq
+   real(rtype), intent(in) :: dt 
+   real(rtype), intent(in) :: odt 
+
+   real(rtype), intent(inout), dimension(kts:kte) :: qr 
+   real(rtype), intent(inout), dimension(kts:kte) :: nr 
+   real(rtype), intent(inout), dimension(kts:kte) :: nr_incld 
+   real(rtype), intent(inout), dimension(kts:kte) :: mu_r 
+   real(rtype), intent(inout), dimension(kts:kte) :: lamr
+   real(rtype), intent(inout) :: prt_liq
+   real(rtype), intent(inout), dimension(kts:kte) :: rflx 
+   real(rtype), intent(inout), dimension(kts:kte) :: qr_tend 
+   real(rtype), intent(inout), dimension(kts:kte) :: nr_tend 
+    
+   logical :: log_qxpresent
+   integer :: k 
+   integer :: kbot, ktop, kdir, k_qxtop, k_qxbot, k_temp 
+   integer :: tmpint1 
+
+   real(rtype) :: dt_left 
+   real(rtype) :: prt_accum 
+   real(rtype) :: Co_max 
+   real(rtype) :: dt_sub
+   real(rtype), dimension(kts:kte) :: V_qr
+   real(rtype), dimension(kts:kte) :: V_nr 
+   real(rtype), dimension(kts:kte) :: flux_qx
+   real(rtype), dimension(kts:kte) :: flux_nx
+   real(rtype) :: fluxdiv_qx, fluxdiv_nx 
+ 
+   real(rtype) :: tmp1, tmp2, dum1, dum2, inv_dum3 , rdumii, rdumjj
+   integer dumii, dumjj 
+
+   ktop = kts
+   kbot = kte
+   kdir = -1 
+   k_qxtop = kbot 
+   log_qxpresent = .false. 
+
+   !find top, determine qxpresent
+   do k = ktop,kbot,-kdir
+      if (qr(k).ge.qsmall) then
+         log_qxpresent = .true.
+         k_qxtop = k
+         exit
+      endif !
+   enddo
+
+   qr_present: if (log_qxpresent) then
+
+      dt_left   = dt  !time remaining for sedi over full model (mp) time step
+      prt_accum = 0._rtype  !precip rate for individual category
+
+      !find bottom
+      do k = kbot,k_qxtop,kdir
+         if (qr(k).ge.qsmall) then
+            k_qxbot = k
+            exit
+         endif
+      enddo
+
+      substep_sedi_r: do while (dt_left.gt.1.e-4_rtype)
+
+         Co_max = 0._rtype
+         V_qr = 0._rtype
+         V_nr = 0._rtype
+
+         kloop_sedi_r1: do k = k_qxtop,k_qxbot,-kdir
+
+            qr_notsmall_r1: if (qr_incld(k)>qsmall) then
+
+               !Compute Vq, Vn:
+               nr(k)  = max(nr(k),nsmall)
+               call get_rain_dsd2(qr_incld(k),nr_incld(k),mu_r(k),lamr(k),     &
+                    tmp1,tmp2,rcldm(k))
+               call find_lookupTable_indices_3(dumii,dumjj,dum1,rdumii,rdumjj,inv_dum3, &
+                    mu_r(k),lamr(k))
+               nr(k) = nr_incld(k)*rcldm(k)
+               !mass-weighted fall speed:
+
+               dum1 = vm_table(dumii,dumjj)+(rdumii-real(dumii))*                       &
+                      (vm_table(dumii+1,dumjj)-vm_table(dumii,dumjj))       !at mu_r
```

Comment:
We had some discussion about how now `mu_r` is constant so this treatment to get `dum1` and `dum2` isn't actually needed.  We suggested to Hugh that it could be deleted but he pointed out that the code may go back to variable `mu_r` in the future so it should remain.  On our end we can decide if we agree or not.  Bringing it up here to invite some discussion.  Main point is that `mu_r` being constant means `dum1`=`dum2`, thus `V_qr`=`dum1` and `dum2` isn't needed.  Also, `vm_table` is much simpler so `dum1` is less complicated than it is currently written.

Reply:
Follow-up, we could make a small subroutine to calculate V_qr and V_nr that keeps this confusing syntax for a future case where `mu_r` does vary, but uses the simpler case for now.  We could use a hard-coded logical to default to the simple case.  Either way won't change the model behavior but just thinking about keeping the code as clean as possible.